src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -49,4 +53,12 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
     startDateTime: LocalDateTime,
     endDateTime: LocalDateTime
   ): List<VisitRestrictionStats>
+
+  @Query(
+    "SELECT v FROM Visit v WHERE v.reference = :reference and v.visitStatus = 'BOOKED' "
+  )
+  fun findBookedVisit(reference: String): Visit?
+
+  @Query("SELECT * FROM visit WHERE reference = :reference AND visit_status = 'RESERVED' ORDER BY modify_timestamp DESC LIMIT 1 ", nativeQuery = true)
```

Comment:
This method might no longer be needed.

Reply:
True, I think it's used in a test so will remove method and changed test